samples/python/django/sample.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+import datetime
+import django
+import pytz
+from django.db import connection
+from django.db.transaction import atomic
+from django.db import transaction
+x = 0
+
+def get_singer(singer_id):
+  global x
+  x += 1
+  return Singer(id=singer_id, first_name='singer', last_name=str(x), full_name= 'singer'+str(x), active=True, created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))
+
+def get_album(album_id, singer=None):
+  global x
+  x += 1
+  return Album(id=album_id, singer=singer, title='album'+str(x), marketing_budget = 200000, release_date = datetime.date.today(), cover_picture= b'hello world',created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))
+
+def get_track(track_id, track_number, album = None):
+  global x
+  x += 1
+  return Track(track_id=track_id, album=album, track_number=track_number, title='track'+str(x), sample_rate=124.543, created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))
+
+def get_venue(venue_id):
+  global x
+  x += 1
+  return Venue(id=venue_id, name='venue'+str(x), description='description'+str(x), created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))
+
+def get_concert(concert_id, venue = None, singer = None):
+  global x
+  x += 1
+  return Concert(id=concert_id, venue=venue, singer=singer, name='concert'+str(x), start_time=datetime.datetime.now(pytz.UTC), end_time=datetime.datetime.now(pytz.UTC)+datetime.timedelta(hours=1), created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))
+
+
+def create_tables():
+  file = open('create_data_model.sql', 'r')
+  ddl_statements = file.read()
+  with connection.cursor() as cursor:
+    cursor.execute(ddl_statements)
+
+@atomic(savepoint=False)
+def test_adding_data():
+  singer = get_singer('1')
+  singer.save()
+
+  album = get_album('1', singer)
+  album.save()
+
+  track = get_track('1', '2', album)
+  track.save(force_insert=True)
```

Comment:
Why do we need force_insert here and not for the others?

Reply:
The reason is because in case of using `save()`, Django will first send `UPDATE` statement and then send an `INSERT` statement if no rows are updated. Now, the `UPDATE` query will look some like `UPDATE tracks set tracks.id = 1, tracks.track_number = '2' ....... where tracks.track_id='1'`. This query will be generated because Django thinks that `track_id` is the primary key but in the actual database combination of `(id, track_number)` is the primary key. So, the database will throw an error something similar to key column cannot be updated.

So, there are two ways to tackle this, either use `force_insert = True` with `save` or use `objects.create()`. The second way will look something like follows:

`Track.objects.create(track_id='1', album=album, track_number='2', title='track'+str(x), sample_rate=124.543, created_at=datetime.datetime.now(pytz.UTC), updated_at=datetime.datetime.now(pytz.UTC))`

Reply:
Ah, thanks for the detailed explanation! But if I understand this correctly, it also means that we can never update a row of an interleaved table, right?

Would setting `Field.editable=False` on the fields fix that? See https://docs.djangoproject.com/en/4.1/ref/models/fields/#editable

Otherwise (and unless there's some other way of marking these fields as non-updateable), I'm afraid that we have to ditch support for interleaved tables in Django. Not being able to update a row seems like too big of a limitation.